satpy/readers/mersi2_l1b.py

```diff
@@ -92,7 +92,7 @@ def get_dataset(self, dataset_id, ds_info):
 
         fill_value = attrs.pop('FillValue', np.nan)  # covered by valid_range
         valid_range = attrs.pop('valid_range', None)
-        if dataset_id['calibration'] == 'counts':
+        if 'calibration' in dataset_id and dataset_id['calibration'] == 'counts':
```

Comment:
The more pythonic way to do this is:
```suggestion
        if dataset_id.get('calibration') == 'counts':
```
IIRC this should work with DataIDs too.

Reply:
Ok